src/parser/jobs/sge/modules/Gauge.tsx

```diff
@@ -0,0 +1,156 @@
+			color: '',
+		},
+		correctHistory: true,
+	}))
+
+	override initialise() {
+		super.initialise()
+
+		const playerFilter = filter<Event>().source(this.parser.actor.id)
+		const partyIds = this.actors.friends.filter(friend => friend.playerControlled).map(friend => friend.id)
+		const partyFilter = filter<Event>().target(oneOf(partyIds))
+		const playerDiagnosisPartyFilter = partyFilter.source(this.parser.actor.id).status(this.data.statuses.EUKRASIAN_DIAGNOSIS.id)
+		this.addEventHook(playerFilter.action(this.data.matchActionId(ADDERSGALL_CONSUMERS)), this.onConsumeAddersgall)
+		this.addEventHook(playerFilter.action(this.data.actions.RIZOMATA.id), () => this.addersgallGauge.generate(1))
+
+		// Hook shield applications/removals that could generate Addersting
+		this.addEventHook(playerDiagnosisPartyFilter.type('statusApply'), this.onShieldApply)
+		this.addEventHook(playerDiagnosisPartyFilter.type('statusRemove'), this.onShieldRemove)
+
+		// Hook shield applications/actions that could prevent Addersting generation
+		this.addEventHook(filter<Event>().type('statusApply').source(noneOf([this.parser.actor.id]))
+			.status(this.data.statuses.EUKRASIAN_DIAGNOSIS.id), this.onShieldOverwrite)
+		this.addEventHook(partyFilter.type('statusApply').status(this.data.matchStatusId(OVERWRITES_DIAGNOSIS)), this.onShieldOverwrite)
+		this.addEventHook(playerFilter.action(this.data.actions.PEPSIS.id), this.onPepsis)
+	}
+
+	/** Addersgall is weird, the timer restarts immediately on death, not after raising, at least in the media tour build */
+	override onDeath(event: Events['death']) {
+		super.onDeath(event)
+
+		this.addersgallTimer.start()
+	}
+
+	//#region Addersgall gauge
+	private onConsumeAddersgall() {
+		this.addersgallGauge.spend(1)
+		if (this.addersgallTimer.expired) {
+			this.addersgallTimer.start()
+		}
+	}
+
+	private onCompleteAddersgallTimer() {
+		this.addersgallGauge.generate(1)
+		if (!this.addersgallGauge.capped) {
+			this.addersgallTimer.refresh()
+		}
+	}
+	//#endregion
+
+	//#region Addersting gauge
+	// Keep track of Eukrasian Diagnosis applications
+	private onShieldApply(event: Events['statusApply']) {
+		this.diagnoses[event.target] = {applyTimestamp: event.timestamp}
+	}
+
+	// Keep track of instances where a Eukrasian Diagnosis was definitely overwritten by another status
+	private onShieldOverwrite(event: Events['statusApply']) {
+		const diagnosis = this.diagnoses[event.target]
+		if (diagnosis == null) { return }
+		diagnosis.consumedOrOverwritten = true
+	}
+
+	// Keep track of shield removals, and set a timestamp hook to resolve the shield (overwrites happen 'after' the removal event but at the same timestamp, so we can't know for sure if the shield was overwritten yet)
+	private onShieldRemove(event: Events['statusRemove']) {
+		const diagnosis = this.diagnoses[event.target]
+		if (diagnosis == null) { return }
+		diagnosis.removeTimestamp = event.timestamp
+		diagnosis.absorbed = event.absorbed
```

Comment:
😬 is there a way we can avoid checking explicit absorbs? they're very loose guesses at best, and if we start consuming them in xiva events, we'll need to make sure that every single possible event source is capable of simulating them to some degree of accuracy.

---

i get the feeling i've had this conversation before lmao. idek

Reply:
At the moment, the behavior of the `absorbed` field is (inherited from FFLogs):
1. A shield applies to the target
2. If the shielded target is the target of any damage events before the shield is removed, the value of the shield is presented as `absorbed` (FFLogs infers the shield got popped by the damage event)
3. If the shielded target is NOT the target of ANY damage events before the shield is removed, the value of the shield is presented as `amount`, which we ignore.

Per discussions with SCH/SGE mentors on the Balance, our current approach with SCH is to track whether a shield got popped or not, but to not consider shields one way or the other with overheal calculations.  So, in the context of this module and whether or not the SGE got a Toxicon stack, the check of whether absorbed is > 0 is necessary.

Reply:
I agree that we should look into a more permanent/log source-agnostic way to detect shield breaks, but that's well above my head at the moment and this was the best "for now" solution I could figure out how to implement. If/when we figure out the core change we'll need, I will _happily_ switch to it in a heartbeat.